### ISSUE TITLE
Empty States

### ISSUE THREAD

**Issue by fritosxii** [matrix-msu/Kora3#58]
Record Results, for project records and form records - No Records:
https://projects.invisionapp.com/d/main#/console/11107725/258859933/preview

Projects- No Projects: 
https://projects.invisionapp.com/d/main#/console/11107725/251340897/preview

Single Project - No Forms:
https://projects.invisionapp.com/d/main#/console/11107725/258858399/preview

Single Form - No Fields: (including hover error when trying to create a new record)
https://projects.invisionapp.com/d/main#/console/11107725/258858398/preview

- [x] Projects - No Projects
- [x] Single Project - No Forms
- [x] Single Form - No Fields
- [x] Single Form - Hover error when trying to create a new record
- [x] Project Record Results - No Records
- [x] Form Record Results - No Records
- [x] Dashboard Empty State
- [x] Record Presets
- [x] Field Presets
- [x] Record Revisions
- [x] Form Association Permissions
- [x] Backups
- [x] Tokens

**Comment by suuuth** [comment]
on branch: suth-empty-states-58

**Comment by austintruchan** [comment]
## Form Page
Update to have the proper text / wording 
Add and use a new icon. Message me when you're ready to recieve it, as github doesn't support svg uploads for some reason. or Download from InVision.
"Create a new form to get started" text should be font-size: 14px
Create a new form should link to the new form page.

**Comment by austintruchan** [comment]
## Single Form No Fields

- [ ] Update to have the proper text / wording
- [ ] Add and use a new icon. Message me when you're ready to recieve it, as github doesn't support svg uploads for some reason. or Download from InVision.
- [ ] Create a new field should link to the new field page.

**Comment by austintruchan** [comment]
I think for the most part these changes above will more less apply to the other empty state pages. However, I plan on mocking up all empty states and preparing proper copy for each sometime soon if you wanted to wait.

**Comment by austintruchan** [comment]
Record Preset Management Empty State: https://projects.invisionapp.com/d/main#/console/11107725/242539674/preview

**Comment by austintruchan** [comment]
Field Value Preset Empty State:
https://projects.invisionapp.com/d/main#/console/11107725/304641459/preview

**Comment by austintruchan** [comment]
Record Revisions Management: https://projects.invisionapp.com/d/main#/console/11107725/304643015/preview

**Comment by austintruchan** [comment]
Request Form Association:
https://projects.invisionapp.com/d/main#/console/11107725/304645070/preview

Create Form Association:
https://projects.invisionapp.com/d/main#/console/11107725/304645656/preview

**Comment by austintruchan** [comment]
Backup Management: https://projects.invisionapp.com/d/main#/console/11107725/304648941/preview

**Comment by austintruchan** [comment]
Token Management: https://projects.invisionapp.com/d/main#/console/11107725/304649546/preview

**Comment by austintruchan** [comment]
- [ ] Looking good. But tHe "Create NEw ..." button for empty states that have it, should be ubuntu medium font. 

- [ ] Also,  remove the default browser active state that this button shows after selecting it.

**Comment by austintruchan** [comment]
Hmmm. Not seeing those changes!

**Comment by austintruchan** [comment]
- [ ] Should be ubuntu-medium font for the 'create new' buttons

- [ ] Can we add the link hover underline animation thing to the 'create new' buttons?

- [ ] On pages that are too short, the bottom section of the empty state will be too close to the bottom of the browser. Add some space below the empty state div to give it some space away from the bottom of the browser, for when the browser is too short.

**Comment by austintruchan** [comment]
- [ ] Use updated token icon found on: https://projects.invisionapp.com/d/main#/console/11107725/304649546/preview

- [ ] No Global search empty state that I could see. Will be based on: https://projects.invisionapp.com/d/main#/console/11107725/258859933/preview

- [ ] For the project and form records page, update icon to be that found here: https://projects.invisionapp.com/d/main#/console/11107725/258859933/preview

- [ ] Project records search page, should have the "No Results" Header similar to the form record page. 

- [ ] Add tooltip to Create New Record button on single form page: https://projects.invisionapp.com/d/main#/console/11107725/258858398/preview

There are a few pages that I can't really test, like all projects, backups, and record presets, so just make sure those have the right icons.

**Comment by suuuth** [comment]
when I try to add the token icon I keep getting this `Strokes get ignored when generating fonts. 
You can convert them to fills to prevent this.` error.  Any idea how to avoid this?

**Comment by austintruchan** [comment]
Ahh, thats my bad, I'll convert it to a fill for you.

**Comment by austintruchan** [comment]
- [ ] Looks like the search results empty state icon is clipping. I readded a fixed one to InVision so we can see if that works. 
https://projects.invisionapp.com/d/main#/console/11107725/258859933/inspect

- [ ] Reduce size of token empty state icon to fontsize 60px.

- [ ] Form results page has multiple "No Results" Headers

- [ ] Not seeing an icon on the field value preset page, on record revisions page, record presets or on association pages. 

- [ ] Hover error for creating a new record on single form page doesn't need to be that tall, can be the same height as other tooltips.

**Comment by suuuth** [comment]
- [x] field value preset 
- [ ] record revisions
- [ ] record presets
- [ ] association

**Comment by austintruchan** [comment]
- [ ] Don't have a hover effect on the create new record button, or be clickable / refresh the page.

**Comment by austintruchan** [comment]
Okay. Lets push this!

**Comment by fritosxii** [comment]
Two things
1) When there are no records the first time you visit the Form Records page, the search filters and the Try Another Search shouldn't be visible. They should only appear after a search has happened. Also, these buttons/filters shouldn't be committed anyway until they are functional, but for note:

- Shouldn't appear first visit with records (currently they don't) (good)
- Should appear after keyword search visit with records (currently they don't) (bad)
- Shouldn't appear first visit with 0 records (currently they do) (bad)
- Should appear after keyword search visit with 0 records (currently they do) (good)

Currently clicking the Try Another Search button, opens up the advanced search view which breaks if there are no advanced search boxes, and does not scroll up to the keyword search box

<img width="1430" alt="screen shot 2018-07-26 at 1 29 54 pm" src="https://user-images.githubusercontent.com/4040076/43278109-132055ba-90d8-11e8-9cce-7870b1afc08d.png">
<img width="1420" alt="screen shot 2018-07-26 at 1 30 04 pm" src="https://user-images.githubusercontent.com/4040076/43278110-1339af60-90d8-11e8-8c16-6568803e31f4.png">

**Comment by suuuth** [comment]
removing `keywords` section from this branch as these were created, styled, and made functional in #88

**Comment by suuuth** [comment]
also going to remove the lower `try another search` button from this branch, as it was also created in #88.  Can we reopen #88, so I can edit that branch to make this button appear only once a search has been made?

As for the no-results icon and `no results found!` text section, should this appear on first page visit, or only once a search has been made, and there are no results found?

**Comment by suuuth** [comment]
- [ ] `no-search-results` appears only once a search has been made, and no results are found
- [ ] `no-records` appears if the user visits the search-form-records page, and there a no records to search through

**Comment by suuuth** [comment]
- [ ] Should appear after keyword search visit with records (currently they don't) (bad)
- [ ] Shouldn't appear first visit with 0 records (currently they do) (bad)

**Comment by austintruchan** [comment]
When a form records page doesn't have any records when you go to it, I would still like the empty state graphic to appear, just without the search summaries, and the try another search button.

If a user does do a search and no results appear, have the search summary section there would be good.

If a user does do a search and gets results, we still need the search summary, and the try another search button there.

**Comment by suuuth** [comment]
- re-added no-results section to the record-index page
- re-added the lower 'try another search' button to the no-results section
- this lower try-another button does not open the advanced search options like it was earlier, it's simply not at all functional currently
- this lower try-another button is not centered, let's push this branch, and create a new issue so that I can style and make this button functional with code from #88 to avoid redundant sections of code, as well as get these two branches working in tandem

**Comment by austintruchan** [comment]
Okay. Yes, lets push and figure out the remaining issues on a new issue in a new branch, instead of having two issues and two branches modifying the same thing.
